server/channels/app/file.go

```diff
@@ -841,7 +841,7 @@ func (t *UploadFileTask) preprocessImage() *model.AppError {
 		return t.newAppError("api.file.upload_file.large_image_detailed.app_error", http.StatusBadRequest)
 	}
 
-	t.fileinfo.HasPreviewImage = true
+	t.fileinfo.HasPreviewImage = false
```

Comment:
This is the right general idea, but we still want to leave `HasPreviewImage` as true for images that aren't animated gifs or animated PNGs. In particular, we still want static PNGs to use a preview image to reduce the amount of bandwidth used by them.

Unfortunately, since animated PNGs are less common than animated gifs, the built-in library doesn't seem to have anything we can use to detect those, and the one Go library I found had an incompatible license with ours.

The best I found was that we might be able to detect the `acTL` chunk in the raw bytes of the image, but perhaps you could find something better for how to identify animated PNGs?

Reply:
@hmhealey, yes i came across this approach. I think acTL chunk is the popular way of determining if its a png or apng. I'll try this approach to see if it works or not.

Reply:
@hmhealey, not sure how I can detect the chunks in GO, I am trying to understand chunks and raw bytes in JS first, but still unable to check for acTL chunk. Any ideas?